source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc

```diff
@@ -74,7 +74,7 @@ void GrpcClientImpl::onFailure(Grpc::Status::GrpcStatus status, const std::strin
   ASSERT(status != Grpc::Status::GrpcStatus::Ok);
   Response response{};
   response.status = CheckStatus::Error;
-  response.status_code = Http::Code::Forbidden;
+  response.status_code = Http::Code::ServiceUnavailable;
```

Comment:
As per our conversation in GH, please make ServiceUnavailable configurable and default it to false. This way we avoid breaking changes.

Reply:
Ah... I thought you said it's okay to hard code 502 as the response :) Seems it's getting more complex than what I thought.

Any idea on the name of the configuration? Any guidance/docs on introducing new configs?

Reply:
Sorry, I misread your last comment about hardcode the status. Unfortunatelly, we can't do that. I think something like `status_code_on_error`, and then let the user decide which status they want would be sufficient. 
Normally, the filter config is done here: https://github.com/envoyproxy/envoy/blob/741df7aa13ddb15acbc77468579f0c6b58542839/api/envoy/config/filter/http/ext_authz/v2/ext_authz.proto#L20

Reply:
Thanks for the pointer. I searched for the code of "use_alpha" in "ext_authz" - feeling like I'm less confident of making the change myself. Do you mind working on top of the current change? I'm more than happy to code review your change. @gsagula

Reply:
I would mind doing, but I can't promise when I will have time to tackle it.

Reply:
Cool! We are blocked by this while integrating Envoy with our (Twilio) gateway. Your help is greatly appreciated!

Reply:
@gsagula do you actually think an enum can work well for `status_code_on_error`? Don't want users to abuse the system by returning like 200, when the ext_authz is down.